pkg/controller/subscription/subscription_controller.go

```diff
@@ -289,6 +279,12 @@ func (r *ReconcileSubscription) HandleUpgrades(ctx context.Context, rhmiSubscrip
 	if !isServiceAffecting || canUpgradeNow {
 		eventRecorder := r.mgr.GetEventRecorderFor("RHMI Upgrade")
 
+		config.Status.UpgradeAvailable = nil
+
+		if err := r.client.Status().Update(context.TODO(), config); err != nil {
+			return reconcile.Result{}, err
+		}
+
```

Comment:
```suggestion
		if config.Status.UpgradeAvailable != nil && config.Status.UpgradeAvailable.TargetVersion == rhmiSubscription.Status.CurrentCSV {
				config.Status.UpgradeAvailable = nil
				if err := r.client.Status().Update(context.TODO(), config); err != nil {
					return reconcile.Result{}, err
				}
		}
```

We only clear the UpgradeAvailable field when the upgrade that has been approved is the one set in the target version. Otherwise then approving a patch release we will lose the upgrade information from the next release